waku/node/waku_node.nim

```diff
@@ -984,10 +998,10 @@ proc mountRlnRelay*(node: WakuNode,
   let rlnRelay = rlnRelayRes.get()
   let validator = generateRlnValidator(rlnRelay, spamHandler)
 
-  # register rln validator for all subscribed relay pubsub topics
-  for pubsubTopic in node.wakuRelay.subscribedTopics:
-    debug "Registering RLN validator for topic", pubsubTopic=pubsubTopic
-    node.wakuRelay.addValidator(pubsubTopic, validator)
+  # register rln validator as default validator
+  debug "Registering RLN validator"
+  node.wakuRelay.addDefaultValidator(validator, "RLN validation failed")
```

Comment:
Is this change out of the scope of the PR?

Reply:
Yes, this is out of scope of this PR. This was already merged. I basically had 2 PRs, one adding the functionality of default validators and using it for RLN, and this one using the validators to check messages in the REST handlers.

So this PR came on top of what was built on the other one. Probably it would have made more sense to open this PR once the other one was merged. Because the other one was small, I had both open in parallel to make things faster, but might have been better to wait for the other one to be merged before opening this one.